app/welcome.js

```diff
@@ -6,11 +6,53 @@
  * Description:
  * Main script for initiating the welcome  page.
  */
- 
- define([ 'jquery', 'bootstrap'], function ($, bootstrap) {
 
-	$("#menu-toggle").click(function(e) {
+define(['jquery', 'app/database', 'bootstrap'], function ($, db, bootstrap) {
+
+	$("#menu-toggle").click(function (e) {
 		e.preventDefault();
 		$("#wrapper").toggleClass("toggled");
 	});
+
+	function createSidebarElements() {
+		var li1 = "<li class=\"sidebar_li\" subject_id=\"";
+    var li2 = "\" language_id=\"";
+		var li3 = "\"><a  href=\"#\"  >";
+		var li4 = "<br><span class=\"sidebar_item\" >for ";
+    var li5 = " speakers</span></a></li>\n";
+		var rows = db.getUnique('getModules', 'subject_name', []);
+		var sidebar = "";
+
+		for (var i = 0; i < rows.length; i++) {
+			sidebar += li1 + rows[i].subject_id + li2 + rows[i].language_id + li3 + rows[i].subject_name + li4 + rows[i].language_name + li5;
+		}
+
+		var container = document.getElementById("sidebar_ul");
+		container.innerHTML = sidebar;
+	}
+
+	function createDatasetsGrid(subjectid, languageid) {
+		var grid = "";
+		var rows = db.getQuery('getDatasets', [languageid, subjectid]);
+		var gridItem1 = "<div class=\"col-md-3 col-sm-6\">\n<div class=\"btn mybutton\" id=\"";
```

Comment:
This is a very interesting idea. But because it will be a significant overhaul of our current design, I think it's better to discuss these among all team members rather than I change it on my own. I therefore made a Trello card here, and we will work on it soon: https://trello.com/c/hKrKaPMu